tests/TemporaryFileTest.php

```diff
@@ -7,6 +7,21 @@
 
 class TemporaryFileTest extends TestCase
 {
+    private string $defaultDirectoryPermissions;
+
+    /**
+     * Setup the test environment.
+     */
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $path = FileHelper::absolutePath('rights-test-permissions', 'local');
+        mkdir($path);
+        $this->defaultDirectoryPermissions = substr(sprintf('%o', fileperms($path)), -4);
+        @rmdir($path);
```

Comment:
May I ask why do you use error suppressor in tests?

Reply:
Because removing the directory doesn't affect the success or failure of the test itself.
I try to remove the testing directory to keep the temporary testing directory empty when the tests are finished.

So it would be fine to remove that line, but I like it keeps things cleaner.

Reply:
I don't go into logic (i.e. resulting removed directory). I am questioning using error suppressing symbol (`@`) in programming and particularly in testing.

Reply:
Successful or a failure when removing the directory does not influence the purpose or result of the test. The code is executed during setup of tests, not the tests themselves.
The purpose is to get the default directory permissions when _creating_ a directory as the user who conducts the tests.